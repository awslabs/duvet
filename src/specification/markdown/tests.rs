// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

use super::*;

fn tokens(contents: &str) -> Vec<Token> {
    Lex::new(contents).collect()
}

macro_rules! snapshot {
    ($name:ident, $contents:expr) => {
        mod $name {
            #[test]
            fn tokens() {
                insta::assert_debug_snapshot!("tokens", super::tokens($contents));
            }

            #[test]
            fn tree() {
                insta::assert_debug_snapshot!("tree", super::parse($contents));
            }
        }
    };
}

snapshot!(
    simple,
    r#"
# This is a test

Content goes here. Another
sentence here.
"#
);

snapshot!(
    multi_line_header,
    r#"
Foo *bar
baz*
======

Content goes here. Another
sentence here.
"#
);

snapshot!(
    multiple,
    r#"
# This is a test

Content goes here. Another
sentence here.

## This is another test

More content goes here

### Nested section

Testing 123

## Up one

Another section
"#
);

snapshot!(
    list_example,
    r#"
# List example

Here is a list:
* Item 1
* Item 2
  * Item 2.1
* Item 3
  * Item 3.1
    * Item 3.1.1
    * Item 3.1.2
  * Item 3.2

Here is a numbered list:
1. Item 1
2. Item 2
3. Item 3

Here is a list with content:
* Item
  More content

  Other content

  * Testing

    Other test

Testing 123

* Item
More content
"#
);
