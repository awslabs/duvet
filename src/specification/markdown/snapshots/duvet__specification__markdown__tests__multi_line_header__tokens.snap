---
source: src/specification/markdown/tests.rs
expression: "super::tokens(r#\"\nFoo *bar\nbaz*\n======\n\nContent goes here. Another\nsentence here.\n\"#)"
---
[
    Line(
        Str {
            value: "",
            pos: 0,
            line: 1,
        },
    ),
    Header {
        line: Str {
            value: "Foo *bar\nbaz*",
            pos: 1,
            line: 2,
        },
        name: Str {
            value: "Foo *bar\nbaz*",
            pos: 1,
            line: 2,
        },
        fragment: None,
        level: 1,
    },
    Line(
        Str {
            value: "",
            pos: 22,
            line: 5,
        },
    ),
    Line(
        Str {
            value: "Content goes here. Another",
            pos: 23,
            line: 6,
        },
    ),
    Line(
        Str {
            value: "sentence here.",
            pos: 50,
            line: 7,
        },
    ),
]
